extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/OwBaseThingHandler.java

```diff
@@ -0,0 +1,264 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.onewire.internal.handler;
+
+import static org.eclipse.smarthome.binding.onewire.internal.OwBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.binding.onewire.internal.OwDynamicStateDescriptionProvider;
+import org.eclipse.smarthome.binding.onewire.internal.OwException;
+import org.eclipse.smarthome.binding.onewire.internal.device.AbstractOwDevice;
+import org.eclipse.smarthome.binding.onewire.internal.device.OwSensorType;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OwBaseThingHandler} class defines a handler for simple OneWire devices
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public abstract class OwBaseThingHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(OwBaseThingHandler.class);
+
+    protected int sensorCount;
+
+    protected final List<AbstractOwDevice> sensors = new ArrayList<AbstractOwDevice>();
+    protected final List<String> sensorIds = new ArrayList<String>();
+    protected long lastRefresh = 0;
+    protected long refreshInterval = 300 * 1000;
+    protected boolean validConfig = false;
+
+    protected @Nullable OwDynamicStateDescriptionProvider dynamicStateDescriptionProvider;
+
+    public OwBaseThingHandler(Thing thing,
+            @Nullable OwDynamicStateDescriptionProvider dynamicStateDescriptionProvider) {
```

Comment:
This is a static mandatory reference in your factory, so how can it be `null` here?

Reply:
Don't know. Will check.

Reply:
fixed